src/util_gen.cpp

```diff
@@ -470,6 +477,14 @@ bool Util::isValidFd(int fd)
   return _real_fcntl(fd, F_GETFL, 0) != -1;
 }
 
+// On failure, returns 0, with errno == EFAULT
+bool Util::isReadableMemoryAddress(const char *addr) {
+  // FIXME: We wanted to call libc:open(), but _real_open goes to second wrapper
+  int fd = _real_syscall(SYS_open, "/dev/null", O_WRONLY);
+  int rc = write(fd, addr, 1);
```

Comment:
Instead of doing an explicit open followed by write, you can do the following:
```int rc = write(PROTECTED_STDERR_FD, addr, 0);
```

It would return `0` if successful, and `-1` with `errno` set to `EFAULT`. This would reduce the unnecessary overhead of performing an open (performing `open` has all sorts of side-effects and let's avoid if we can).